### PR TITLE
refactor: make it easier to deal with commit conflict

### PR DESCRIPTION
**Pull request by SkyFan2002**
I hereby agree to the terms of the CLA available at: https://databend.rs/dev/policies/cla/

## Summary

Summary about this PR

- Closes #issue

### CODE REVIEW

**Comment by dantengsky** [comment]
## Summary(By llmchain.rs)
* **Simplify Delete Interpreter**
The interpreter.rs file was modified to remove a match statement and replace it with a simple assignment. The interpreter_delete.rs file was deleted, removing several import statements, variable creations, and function calls related to the GlobalIORuntime and TableLockHandlerWrapper.

* **Add MutationKind Parameter**
The pipeline_builder.rs file was modified to add an import statement for MutationKind and a MutationKind parameter to the build_row_fetcher_pipeline function call. The recluster.rs, replace.rs, and update.rs files were modified to add an import statement for MutationKind and a mutation_kind parameter to the chain_mutation_pipes function call.

* **Add Conflict Resolution Tests**
The conflict.rs file was added to the project to contain tests for conflict resolution in the Fuse storage engine. The tests check if a delete operation can be applied when the segment to be deleted has been modified by other operations and if a delete operation can be applied when the segment to be deleted is still present in the latest snapshot.

* **Modify Mutation Accumulator**
The mutation_accumulator.rs file was modified to add several structs/enums, a kind field to the MutationAccumulator struct, and modifications to the new and apply functions to handle different MutationKind values and create ConflictResolveContext accordingly. The CommitMeta struct was modified to have a conflict_resolve_context field instead of segments and summary fields.

* **Modify Snapshot Generator**
The snapshot_generator.rs file was modified to add several structs/enums, methods to ConflictResolveContext to detect conflicts and merge segments, and modifications to MutationGenerator and AppendGenerator to use ConflictResolveContext instead of merged_segments and merged_statistics. The check_fill_default method in AppendGenerator was modified to return a Result<bool> instead of a bool.

**Comment by drmingdrmer** [comment]
### PR summary(by Claude):


```

File: src/query/service/src/interpreters/interpreter_delete.rs [hunk 1]
```diff
@@ -96,17 +94,6 @@ impl Interpreter for DeleteInterpreter {
         let catalog_name = self.plan.catalog_name.as_str();
         let db_name = self.plan.database_name.as_str();
         let tbl_name = self.plan.table_name.as_str();
-
-        let tbl = self.ctx.get_table(catalog_name, db_name, tbl_name).await?;
-        let table_info = tbl.get_table_info().clone();
-
-        // Add table lock heartbeat.
-        let handler = TableLockHandlerWrapper::instance(self.ctx.clone());
```
Comment (zhyass) [inline src/query/service/src/interpreters/interpreter_delete.rs:104]: This pr can reduce some commit conflicts. However,  it cannot avoid the inevitable conflicts. I think add table lock is still necessary. Unless we support segment or block level locks.

**Comment by SkyFan2002** [comment]
Some [pressure tests](https://github.com/SkyFan2002/databend/tree/refactor_long_run/tests/test_conflict) have been performed for this PR: deletion and compaction were executed concurrently. After that, system metrics and remaining data were examined, which showed that many conflicts between deletion and compaction were resolved.

**Comment by dantengsky** [comment]
> Some [pressure tests](https://github.com/SkyFan2002/databend/tree/refactor_long_run/tests/test_conflict) have been performed for this PR: deletion and compaction were executed concurrently. After that, system metrics and remaining data were examined, which showed that many conflicts between deletion and compaction were resolved.

 :+1: thanks

**Review by dantengsky — APPROVED** [review]

**Review by zhyass — APPROVED** [review]

**Review by zhyass — APPROVED** [review]

**Pull request merged**